src/mmw/js/src/core/weatherStationLayer.js

```diff
@@ -0,0 +1,47 @@
+"use strict";
+
+var L = require('leaflet'),
+    Backbone = require('../../shim/backbone'),
+    Marionette = require('../../shim/backbone.marionette'),
+    weatherStationPopupTmpl = require('./templates/weatherStationPopup.html');
+
+var Layer = {
+    createLayer: function(geojsonFeatureCollection) {
+        return L.geoJson(JSON.parse(geojsonFeatureCollection), {
+            pointToLayer: function (feature, latlng) {
+                return L.circleMarker(latlng, {
+                    fill: true,
+                    fillColor: 'steelblue',
+                    fillOpacity: 0.2
+                });
+            },
+
+            onEachFeature: function(feature, marker) {
+                var model = new Backbone.Model(feature.properties),
+                    view = new WeatherStationPopupView({ model: model });
+
+                marker.bindPopup(view.render().el, { className: 'data-catalog-popover'});
+            },
+        });
+    }
+};
+
+var WeatherStationPopupView = Marionette.ItemView.extend({
+    template: weatherStationPopupTmpl,
+    className: 'weather-station-popup',
+
+    templateHelpers: function() {
+        return {
+            station_id: this.model.get('station_id'),
```

Comment:
The attributes of `this.model` are available natively to the template, so this entire `templateHelpers` function can be elided.